packages/cli-repl/src/mongosh-repl.ts

```diff
@@ -131,6 +133,53 @@ type Mutable<T> = {
   -readonly [P in keyof T]: T[P];
 };
 
+function filterStartingWith({
+  kind,
+  name,
+  trigger,
+}: {
+  kind: string;
+  name: string;
+  trigger: string;
+}): boolean {
+  name = name.toLocaleLowerCase();
+  trigger = trigger.toLocaleLowerCase();
+
+  /*
+  1. If the trigger was blank and the kind is not property/method filter out the
+     result. This way if you autocomplete db.test.find({ you don't get all the
+     global variables, just the known collection field names and mql but you can
+     still autocomplete global variables and functions if you type part of the
+     name.
+  2. Don't filter out exact matches (where filter === name) so that we match the
+     behaviour of the node completer.
+  3. Make sure the name starts with the trigger, otherwise it will return every
+     possible property/name that's available at that level. ie. all the "peer"
+     properties of the things that match.
+  */
+  //console.log(name, kind);
+  // TODO: This can be improved further if we first see if there are any
```

Comment:
Filed a ticket to just improve the filtering: https://jira.mongodb.org/browse/MONGOSH-2207